.github/workflows/update_api.yml

```diff
@@ -0,0 +1,76 @@
+name: Update API 
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "30 3 * * *"
+jobs:
+  update-api:
+    if: ${{ github.repository == 'opensearch-project/opensearch-py' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - name: Config git to rebase
+        run: git config --global pull.rebase true
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run: |
+          python3.7 -m pip install nox
+      - name: Generate API
+        run: nox -s generate
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
+      - name: Create pull request
+        id: cpr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: Updated opensearch-py to reflect the latest OpenSearch API spec (${{ steps.date.outputs.date }})
+          title: Updated opensearch-py to reflect the latest OpenSearch API spec
+          body: |
+            Updated [opensearch-py](https://github.com/opensearch-project/opensearch-py) to reflect the latest [OpenSearch API spec](https://github.com/opensearch-project/opensearch-api-specification/blob/main/OpenSearch.openapi.json)
+            Date: ${{ steps.date.outputs.date }}
+          branch: automated-api-update
+          base: main
+          committer: opensearchpy-api-update-bot <noreply@github.com>
+          author: opensearchpy-api-update-bot <noreply@github.com>
+      - name: Check out update branch
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        run: |
+          git fetch origin automated-api-update
+          git checkout automated-api-update
+      - name: Update CHANGELOG
+        uses: jacobtomlinson/gha-find-replace@v3
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        with:
+          find: "- Your contribution here."
+          replace: "- Updated opensearch-py to reflect the latest OpenSearch API spec ([#${{steps.cpr.outputs.pull-request-number}}](https://github.com/opensearch-project/opensearch-py/pull/${{steps.cpr.outputs.pull-request-number}}))\n- Your contribution here."
+          include: "**CHANGELOG.md"
+      - name: Commit and Push
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        run: |
+          git config --local user.name 'opensearchpy-api-update-bot'
```

Comment:
Why is this step needed if the pull request is already created?

Reply:
updating changelog

Reply:
Also, you can use a github action to do this instead of doing it manually. See https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/dependabot_pr.yml#L56